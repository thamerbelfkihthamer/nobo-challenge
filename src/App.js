import React, {Component} from 'react';
import './App.css';
import Films from './components/Films';
import Layout from './components/Layout';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import Film from './components/Film';
import NoMatch from './components/NoMatch';

export default class App extends Component {

  render() {
    return (
      <div className="App">
         <Router>
            <Layout>
              <Switch>
                <Route exact path="/" component={Films} />
                <Route exact path="/films" component={Films} />
                <Route path="/films/:id" exact component={Film} />
                <Route component={NoMatch} />
              </Switch>
            </Layout>
          </Router>
      </div>
    )
  }
}
