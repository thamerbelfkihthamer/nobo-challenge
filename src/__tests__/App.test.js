import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    test('has a valid snapshot', () => {
        const app = renderer.create(<App />);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    })
})