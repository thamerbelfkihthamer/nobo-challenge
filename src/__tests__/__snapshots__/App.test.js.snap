// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div>
      <header
        className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionRelative"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
          >
            <div
              className="MuiAvatar-root"
            >
              <img
                alt="Nobo"
                className="MuiAvatar-img"
                src="logo-popup-250x150.png"
              />
            </div>
          </h6>
          <div
            style={
              Object {
                "fontSize": "16px",
                "marginLeft": "20px",
              }
            }
          >
            <a
              href="/films"
              onClick={[Function]}
            >
              Films
            </a>
          </div>
        </div>
      </header>
    </div>
    <main>
      <div
        style={
          Object {
            "marginBottom": 100,
            "marginTop": 50,
          }
        }
      >
        <div
          className="MuiContainer-root MuiContainer-maxWidthSm"
        >
          <form
            onSubmit={[Function]}
            style={
              Object {
                "marginBottom": 50,
              }
            }
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              style={
                Object {
                  "margin": 8,
                }
              }
            >
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-describedby="outlined-full-width-helper-text"
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  disabled={false}
                  id="outlined-full-width"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="What's the tv show you are looking for ? "
                  required={false}
                  type="text"
                  value="test"
                />
                <div
                  className="MuiInputAdornment-root"
                >
                  <button
                    className="MuiButtonBase-root MuiIconButton-root"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-416 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-417"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                id="outlined-full-width-helper-text"
              >
                ex : Fast Layne, Go Jetters ...
              </p>
            </div>
          </form>
        </div>
      </div>
      <div
        className="MuiContainer-root MuiContainer-maxWidthMd"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
          style={
            Object {
              "marginBottom": 100,
            }
          }
        />
      </div>
    </main>
    <footer
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Nobo
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright Â© 
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Made with Love
        </a>
         
        2019
        .
      </p>
    </footer>
  </div>
</div>
`;
